### PR TITLE
Only import PII constants during Curator import

### PR DESCRIPTION
**Pull request by ayushdg**
One approach to fixing #59. 
The root cause seems to come from the fact that importing space (which imports and calls a cupy cuda function) somehow impacts the state of the system and prevent a cluster starting up using all available GPUs on the machine. 

Currently the only imports in Curator that lead to this situation is importing `DEFAULT_LANGUAGE` from the pii modules which transitively ends up importing presidio->spacy.

This pr moves these constants to a separate file so that we don't end up importing all other dependencies during Curator import.

### CODE REVIEW

File: nemo_curator/modifiers/pii_modifier.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 import pandas as pd
 
 from nemo_curator.modifiers import DocumentModifier
-from nemo_curator.pii.algorithm import DEFAULT_LANGUAGE
+from nemo_curator.pii.constants import DEFAULT_LANGUAGE
```
Comment (VibhuJawa) [inline nemo_curator/modifiers/pii_modifier.py:20]: n.p. 

Why not import it DEFAULT_MAX_DOC_SIZE here, if its just a constant? 

```suggestion
from nemo_curator.pii.constants import DEFAULT_LANGUAGE, DEFAULT_MAX_DOC_SIZE
```

File: nemo_curator/pii/algorithm.py [hunk 2]
```diff
@@ -30,36 +30,16 @@
 from presidio_anonymizer import AnonymizerEngine, BatchAnonymizerEngine
 from presidio_anonymizer.entities import OperatorConfig
 
+from nemo_curator.pii.constants import DEFAULT_LANGUAGE, SUPPORTED_ENTITIES
 from nemo_curator.pii.custom_batch_analyzer_engine import CustomBatchAnalyzerEngine
 from nemo_curator.pii.custom_nlp_engine import CustomNlpEngine
 from nemo_curator.pii.recognizers.address_recognizer import AddressRecognizer
```
Comment (VibhuJawa) [inline nemo_curator/pii/algorithm.py:37]: Will suggest adding a comment here (and creating an issue) that importing this module leads to cuda context problems when used with dask_cuda.

**Review by VibhuJawa — APPROVED** [review]
Mostly looks good to me, nitpicks around  comments and code structure. 

Thanks a ton for fixing this for us.

File: nemo_curator/pii/algorithm.py (hunk 2 above)
Comment (ayushdg) [reply nemo_curator/pii/algorithm.py:41]: Opened #64 and added comment

**Comment by VibhuJawa** [comment]
Tested locally, works, please go ahead and merge.

**Pull request merged**